infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/fsm/NegotiatingState.java

```diff
@@ -278,6 +278,14 @@ private void processNegotiationResponse(LogReplicationMetadataResponseMsg negoti
         if (negotiationResponse.getSnapshotStart() == negotiationResponse.getSnapshotTransferred() &&
                 negotiationResponse.getSnapshotStart() == negotiationResponse.getSnapshotApplied() &&
                 negotiationResponse.getLastLogEntryTimestamp() >= negotiationResponse.getSnapshotStart()) {
+
+            /*
+             * In the event during the last snapshot sync, apply had processed on the sink, but active had
+             * been interrupted before updating the replication metadata then we should update the prior sync's info.
+             */
+            fsm.getSourceManager().getAckReader().setBaseSnapshot(negotiationResponse.getSnapshotApplied());
```

Comment:
I shared a prior bug offline where this was hit and we decided to not address it.  Lets discuss if/how to add a fix.

Updating the status does not preserve the accuracy of prior snapshot sync.  It will also set it to complete every time there is a transition to LogEntry sync on a normal restart.

Reply:
decided to make this fix as per offline discussion.